### PR TITLE
Python not async

### PR DESCRIPTION
**Pull request by necrophonic**
**NOT TO MERGE**

Opening to allow commenting

### CODE REVIEW

File: publisher.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    ftp.close()
+
+
+def main():
+    pool = Pool(10)
+    tasks = []
+    ftp = FTP()
+    ftp.connect(host='127.0.0.1', port=2121)
+    ftp.login(user='ons', passwd='ons')
+    files = (file for file in ftp.nlst() if '.xlsx' in os.path.splitext(file)[1])
```
Comment (necrophonic) [inline publisher.py]: Does this deal with the situation where there are subfolders as well as regular files?

File: publisher.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    ftp = FTP()
+    ftp.connect(host='127.0.0.1', port=2121)
+    ftp.login(user='ons', passwd='ons')
+    with BytesIO() as sf:
+        s.post('http://0.0.0.0:8080/upload/bres/1/{}'.format(match), data=sf)
+        ftp.retrbinary('RETR %s' %match, sf.write)
+    ftp.close()
+
+
+def main():
```
Comment (necrophonic) [inline publisher.py]: Does this run the once or is it on a schedule / polling interval?

**Comment by JamesGardiner** [comment]
@necrophonic and @warrenbailey  - healthcheck added and now running from Flask app. Feel free to review. Performance wise the whole ftp transfer takes place in ~3.5 seconds on my machine with 1000 test files.

File: publisher.py (hunk 2 above)
Comment (JamesGardiner) [reply publisher.py]: Runs on a background thread on an interval. The interval is currently hardcoded for the spike, but could be set using an env var in production.

File: publisher.py (hunk 1 above)
Comment (JamesGardiner) [reply publisher.py]: This is just a wrapper around the ftp `NLST` command. Would be simple to search the whole directory tree using path manipulation in os.

File: publisher.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    r = s.post('http://0.0.0.0:8080/upload/bres/1/{}'.format(file_name),
+                      files={'file': binary_file.getvalue()})
+    binary_file.close()
+    return r
+
+
+def run():
+    pool = Pool(8)
+    tasks = []
+    ftp = FTP()
```
Comment (eldeal) [inline publisher.py]: Could this call the conn method instead of duplicating the code?

Comment (JamesGardiner) [reply publisher.py]: Yep! I'll update

File: settings.py [hunk 4]
```diff
@@ -0,0 +1,5 @@
+import logging
+import os
+
+LOGGING_FORMAT = "%(asctime)s|%(levelname)s: sdx-validate: %(message)s"
```
Comment (eldeal) [inline settings.py]: This isnt sdx-validate :)

**Comment by necrophonic** [comment]
Archiving pull request retaining history for future reference

**Pull request closed without merging**